catalog/data_metastore.go

```diff
@@ -10,15 +12,54 @@ import (
 
 func DataSourceMetastore() common.Resource {
 	type AccountMetastoreByID struct {
-		Id        string                 `json:"metastore_id"`
+		Id        string                 `json:"metastore_id,omitempty" tf:"computed"`
+		Name      string                 `json:"name,omitempty" tf:"computed"`
+		Region    string                 `json:"region,omitempty" tf:"computed"`
 		Metastore *catalog.MetastoreInfo `json:"metastore_info,omitempty" tf:"computed" `
 	}
 	return common.AccountData(func(ctx context.Context, data *AccountMetastoreByID, acc *databricks.AccountClient) error {
-		metastore, err := acc.Metastores.GetByMetastoreId(ctx, data.Id)
-		if err != nil {
-			return err
+		if data.Id == "" && data.Name == "" && data.Region == "" {
+			return fmt.Errorf("one of metastore_id, name or region must be provided")
 		}
-		data.Metastore = metastore.MetastoreInfo
+		if (data.Id != "" && data.Name != "") || (data.Region != "" && data.Id != "") || (data.Region != "" && data.Name != "") {
+			return fmt.Errorf("only one of metastore_id, name or region must be provided")
+		}
+		if data.Id != "" {
+			minfo, err := acc.Metastores.GetByMetastoreId(ctx, data.Id)
+			if err != nil {
+				return err
+			}
+			data.Metastore = minfo.MetastoreInfo
+		} else {
+			metastores, err := acc.Metastores.ListAll(ctx)
+			if err != nil {
+				return err
+			}
+			minfos := []catalog.MetastoreInfo{}
+			if data.Name != "" {
+				for _, v := range metastores {
+					if v.Name == data.Name {
```

Comment:
would it be better to match names without case sensitivity - the metastore name gets converted to all lowercase on the backend

Reply:
ah, yes - that's what I wasn't sure about. Let me fix it.

Reply:
done